components/script/dom/node.rs

```diff
@@ -281,6 +281,16 @@ impl Node {
         }
     }
 
+    #[allow(unsafe_code)]
+    pub fn clean_up_layout_data(&self) {
+        self.owner_doc().cancel_animations_for_node(self);
+        unsafe {
+            self.style_and_layout_data
+                .borrow_for_script_deallocation()
```

Comment:
This should be borrow_mut, right? Then there's no need for unsafe.

Reply:
That's right. I've pushed a new version of the branch that uses `borrow_mut()` as well as fixing some test expectations.